python/Ganga/GPIDev/Credentials/CredentialStore.py

```diff
@@ -83,6 +83,11 @@ def create(self, query, create=True, check_file=False):
             The newly created ICredentialInfo object
         """
 
+        try:
+            assert isinstance(query, ICredentialRequirement)
+        except AssertionError:
+            raise CredentialsError("Object: '%s' is NOT an ICredentialRequirement instance" % str(query))
```

Comment:
I was planning to add the other parameters here and have a more general error at the assertion error but I haven't got around to adding checks for the other parameters yet.